### PR TITLE
CPLP-1847 decline service app

### PR DESCRIPTION
**Pull request by Phil91**
In this PR [CPLP-1847](https://jira.catena-x.net/browse/CPLP-1847) & [CPLP-1519](https://jira.catena-x.net/browse/CPLP-1519) are resolved.

Please review the Database [PR](https://github.com/tractusx-team-portal-onboarding/devenv/pull/186) as well.

### CODE REVIEW

File: src/marketplace/Offers.Library/Service/OfferService.cs [hunk 1]
```diff
@@ -587,6 +587,66 @@ public async Task ApproveOfferRequestAsync(Guid offerId, string iamUserId, Offer
...
 
+    /// <inheritdoc />
+    public async Task DeclineOfferAsync(Guid offerId, string iamUserId, OfferDeclineRequest data, OfferTypeId offerType, NotificationTypeId notificationTypeId, IDictionary<string,IEnumerable<string>> notificationRecipients, string basePortalAddress)
+    {
+        var offerRepository = _portalRepositories.GetInstance<IOfferRepository>();
+        var declineData = await offerRepository.GetOfferDeclineDataAsync(offerId, iamUserId, offerType).ConfigureAwait(false);
+
+        if (declineData == default)
+        {
+            throw new NotFoundException($"{offerType.ToString()} {offerId} does not exist");
```
Comment (ntruchsess) [inline src/marketplace/Offers.Library/Service/OfferService.cs:598]: calling 'ToString()' is redundant

File: src/marketplace/Services.Service/BusinessLogic/ServiceBusinessLogic.cs [hunk 2]
```diff
@@ -26,6 +26,7 @@
 using Org.CatenaX.Ng.Portal.Backend.PortalBackend.DBAccess;
 using Org.CatenaX.Ng.Portal.Backend.PortalBackend.DBAccess.Models;
 using Org.CatenaX.Ng.Portal.Backend.PortalBackend.DBAccess.Repositories;
+using Org.CatenaX.Ng.Portal.Backend.PortalBackend.PortalEntities.Entities;
```
Comment (ntruchsess) [inline src/marketplace/Services.Service/BusinessLogic/ServiceBusinessLogic.cs:29]: please remove redundant usings

File: src/portalbackend/PortalBackend.Migrations/Migrations/20221206074134_CPLP-1519-AddReleaseRejectionNotifications.Designer.cs [hunk 3]
```diff
@@ -0,0 +1,6353 @@
+﻿// <auto-generated />
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.Migrations/Migrations/20221206074134_CPLP-1519-AddReleaseRejectionNotifications.Designer.cs:1]: header comment is missing

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs [hunk 4]
```diff
@@ -1,24 +1,4 @@
...
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations
- * under the License.
- *
- * SPDX-License-Identifier: Apache-2.0
- ********************************************************************************/
-
-// <auto-generated />
+﻿// <auto-generated />
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs]: added

File: src/portalbackend/PortalBackend.Migrations/Migrations/20221206074134_CPLP-1519-AddReleaseRejectionNotifications.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/portalbackend/PortalBackend.Migrations/Migrations/20221206074134_CPLP-1519-AddReleaseRejectionNotifications.cs]: added

File: src/portalbackend/PortalBackend.Migrations/Migrations/20221206074134_CPLP-1519-AddReleaseRejectionNotifications.Designer.cs (hunk 3 above)
Comment (Phil91) [reply src/portalbackend/PortalBackend.Migrations/Migrations/20221206074134_CPLP-1519-AddReleaseRejectionNotifications.Designer.cs]: added

File: src/marketplace/Offers.Library/Service/OfferService.cs [hunk 6]
```diff
@@ -587,6 +587,66 @@ public async Task ApproveOfferRequestAsync(Guid offerId, string iamUserId, Offer
...
+        }
+
+        if (!declineData.IsUserOfProvider)
+        {
+            throw new ArgumentException($"{offerType.ToString()} not found. Either not existing or no permission for change.", nameof(iamUserId));
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/marketplace/Offers.Library/Service/OfferService.cs]: changed. thanks for the hint

File: src/marketplace/Offers.Library/Service/OfferService.cs (hunk 1 above)
Comment (Phil91) [reply src/marketplace/Offers.Library/Service/OfferService.cs]: removed for all callings

**Pull request merged**